yb-voyager/cmd/exportData.go

```diff
@@ -252,41 +250,53 @@ func debeziumExportData(ctx context.Context, tableList []*sqlname.SourceName) er
 			continue
 		}
 		progressTracker.UpdateProgress(status)
-		if status != nil && exportingSnapshot && status.SnapshotExportIsComplete() {
-			exportingSnapshot = false
-			progressTracker.Done(status)
-			createExportDataDoneFlag()
-			err = writeDataFileDescriptor(exportDir, status)
+		if !snapshotComplete {
+			snapshotComplete, err = checkAndHandleSnapshotComplete(status, progressTracker)
 			if err != nil {
-				return fmt.Errorf("failed to write data file descriptor: %w", err)
+				return err
 			}
-			outputExportStatus(status)
-
 		}
 		time.Sleep(time.Millisecond * 500)
 	}
-
 	if err := debezium.Error(); err != nil {
-		return fmt.Errorf("debezium failed during initial snapshot phase: %w", err)
+		return fmt.Errorf("debezium failed with error: %w", err)
 	}
-
-	if !liveMigration && useDebezium {
-		log.Infof("snapshot export is complete, stopping debezium...")
-		return debezium.Stop()
-	}
-
-	// live migration part
-	color.Blue("streaming changes to a local queue file...")
-	for debezium.IsRunning() {
-		time.Sleep(time.Second)
-	}
-	if err := debezium.Error(); err != nil {
-		return fmt.Errorf("debezium failed during live migration phase: %v", err)
+	// handle case where debezium finished before snapshot completion
+	// was handled in above loop
+	if !snapshotComplete {
+		status, err = debezium.GetExportStatus()
+		if err != nil {
+			return fmt.Errorf("failed to read export status: %w", err)
+		}
+		snapshotComplete, err = checkAndHandleSnapshotComplete(status, progressTracker)
+		if !snapshotComplete || err != nil {
+			return fmt.Errorf("snapshot was not completed: %w", err)
+		}
 	}
 
+	log.Info("Debezium exited normally.")
 	return nil
 }
 
+func checkAndHandleSnapshotComplete(status *dbzm.ExportStatus, progressTracker *ProgressTracker) (bool, error) {
+	if status.SnapshotExportIsComplete() {
```

Comment:
It is idiomatic in Go to reduce nesting by returning at the earliest possible https://danp.net/posts/reducing-go-nesting/

```
if !status.SnapshotExportIsComplete() {
  return false, nil
}
progressTracker.Done(status)
...
```